autoscheduler/frontend/src/types.ts

```diff
@@ -1,3 +1,32 @@
+/* eslint-disable import/prefer-default-export */
+
 /**
  * This file is where we will declare the types of each data-holding class we will be using.
  */
+export enum MeetingType {
+  LEC, LAB, RES, INS
+}
+
+export interface Meeting {
+  id: number;
+  crn: number;
+  building: string | null;
+  meetingDays: boolean[];
+  startTimeHours: number;
+  startTimeMinutes: number;
+  endTimeHours: number;
+  endTimeMinutes: number;
+  meetingType: MeetingType;
+  section: Section;
+}
+
+export interface Section {
+  id: number;
+  subject: string;
+  courseNum: number;
+  sectionNum: number;
+  minCredits: number;
+  maxCredits: number | null;
+  currentEnrollment: number;
+  instructor: number;
+}
```

Comment:
Since this is basically the backbone of the front-end, I think this deserves its own PR so we can discuss what all we actually need.

Reply:
Do you want all of the types from `scraper/models.py` implemented? I only made the two that I needed to get the Meeting testing working, but it's trivial to copy the types from the backend branch if you want to get it all done at once

Reply:
. Only downside is there may be things we can't forsee that we'll need when actually using them, but I think it'll be fine if we change them in the future

Reply:
I don't expect it to be a 1:1 to the models(i.e. the department probably won't need its term)